.github/actions/setup-woocommerce-monorepo/action.yml

```diff
@@ -25,21 +37,38 @@ runs:
           uses: 'actions/setup-node@8f152de45cc393bb48ce5d89d36b731f54556e65'
           with:
               node-version-file: '.nvmrc'
-              # We only want to use the cache if something is being installed.
-              cache: ${{ inputs.install != 'false' && 'pnpm' || '' }}
         - name: 'Setup PHP'
           if: ${{ inputs.php-version != 'false' }}
           uses: 'shivammathur/setup-php@a36e1e52ff4a1c9e9c9be31551ee4712a6cb6bd0'
           with:
               php-version: '${{ inputs.php-version }}'
               coverage: 'none'
-        - name: 'Cache Composer Dependencies'
-          if: ${{ inputs.build == 'false' }}
+        - name: 'Cache: identify pnpm caching directory'
```

Comment:
To get it functioning right, `pnpm install` needs to be (try) `pnpm install --filter="TARGET..." --dedupe-peer-dependents=false`. Otherwise, all cached entities are complete deps.